### PR TITLE
Fixing recent CI failures since 23.3.0 release

### LINKED ISSUE #186: Tests failing after libmamba 1.4.2 release
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

The following tests are failing in CI right after we released the 23.3.0 conda-libmamba-solver. It is hard to assign a definitive issue title since I still don't understand the core issue. https://github.com/conda/conda-libmamba-solver/actions/runs/4703354936/jobs/8341750886.

 - [ ] tests/core/test_solve.py::test_force_remove_1 (different package versions Windows/Linux/OSX)
 - [ ] tests/core/test_solve.py::test_aggressive_update_packages (different package versions Windows/Linux/OSX)
 - [ ] tests/core/test_solve.py::test_update_deps_2 (Unsatisfiable error Windows/Linux/OSX)

These errors happen on all OSs. We are only seeing this on conda-forge but this is due to 1.4.2 not being in defaults yet. 

The following unit-tests fail on conda-libmamba-solver

 - [ ] conda-libmamba-solver-src/tests/test_modified_upstream.py::test_freeze_deps_1
 - [ ] conda-libmamba-solver-src/tests/test_modified_upstream.py::test_conda_downgrade
 - [ ] conda-libmamba-solver-src/tests/test_modified_upstream.py::test_fast_update_with_update_modifier_not_set

### PR DESCRIPTION
**Pull request by costrouc**
Investigates #186 

There are several failures in CI that occurred at the same time this is a PR to resolve those issues.

### CODE REVIEW

File: conda_libmamba_solver/mamba_utils.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Copyright (C) 2019 QuantStack and the Mamba contributors.
+# Copyright (C) 2022 Anaconda, Inc
```
Comment (costrouc) [inline conda_libmamba_solver/mamba_utils.py:2]: @jaimergp and @jezdez is it appropriate to add this line since I checked the git blame history and @jaimergp did make modifications after the QuantStack/Mamba work. I also excluded this file since it was causing precommit to fail.

**Comment by costrouc** [comment]
Diving into one of the CI tests which is failing

````shell
CONDA_SOLVER=libmamba pytest /opt/conda-src/tests/core/test_solve.py::test_force_remove_1
````

I can verfiy this passes without the conda solver being specified and using classic. This test looks like it is failing for older versions of conda-libmamba-solver ... leading me to think this may have to do with libmamba. Looking into.

**Comment by costrouc** [comment]
I can reproduce 

```shell
CONDA_SOLVER=libmamba pytest tests/cli/test_main_clean.py::test_clean_all_mock_lstat
```

Getting the following error

```shell
FAILED tests/cli/test_main_clean.py::test_clean_all_mock_lstat[False] - AssertionError: assert 'WARNING:' in ''
```

These are the lines https://github.com/conda/conda/blob/main/tests/cli/test_main_clean.py#L326-L327

**Comment by costrouc** [comment]
I cannot reproduce 

```shell
CONDA_SOLVER=libmamba pytest tests/core/test_subdir_data.py::test_state_is_not_json
````

**Comment by costrouc** [comment]
So to summarize:
 - [X] several tests are failing due to libmambapy 1.4.1 -> 1.4.2 see https://github.com/conda/conda-libmamba-solver/pull/187#issuecomment-1509186913 for summary
 - [ ] tests/cli/test_main_clean.py::test_clean_all_mock_lstat[False] is failing (unsure exactly why at the moment but can reproduce)
 - [ ] tests/core/test_subdir_data.py::test_state_is_not_json is failing but cannot reproduce

**Comment by wolfv** [comment]
Any clues what is going wrong / what changed in libmamba?

**Comment by jaimergp** [comment]
Hm, looking at the changelog, these would be the PRs that touched solver code:

- https://github.com/mamba-org/mamba/pull/2399
- https://github.com/mamba-org/mamba/pull/2433
- https://github.com/mamba-org/mamba/pull/2428

Those are significant changes so I can't tell right away what could be the cause.

**Comment by jaimergp** [comment]
> `CONDA_SOLVER=libmamba pytest tests/test_activate.py::ShellWrapperIntegrationTests::test_dash_basic_integration
`

This one is known to be flaky so it fails here and there, no worries.

File: conda_libmamba_solver/mamba_utils.py (hunk 1 above)
Comment (jaimergp) [reply conda_libmamba_solver/mamba_utils.py:2]: I guess it's ok :)

File: dev/requirements.txt [hunk 2]
```diff
@@ -4,7 +4,9 @@ pip
 boltons>=23.0.0
 conda>=23.3.1
 importlib-metadata
-libmamba>=1.4.1
-libmambapy>=1.4.1
+libmamba==1.4.1
+libmambapy==1.4.1
```
Comment (jaimergp) [inline dev/requirements.txt:8]: Maybe we can leave it like this
```suggestion
libmamba>=1.4.1,!=1.4.2
libmambapy>=1.4.1,!=1.4.2
```

File: dev/requirements.txt [hunk 3]
```diff
@@ -4,7 +4,9 @@ pip
...
 # be explicit about sqlite because sometimes it's removed from the env :shrug:
 sqlite
+# needed for many conda tests
+flask
```
Comment (jaimergp) [inline dev/requirements.txt:12]: Isn't this provided by `conda/conda` requirements?

**Comment by jaimergp** [comment]
Given that we have other work in the pipeline blocked by this, and the failing tests are not core issues (kind of edge cases), I'd say we annotate them in our meta-issue and mark them as `xfail` or `skipif` with a condition for `libmamba==1.4.2`.

File: dev/requirements.txt (hunk 2 above)
Comment (jaimergp) [reply dev/requirements.txt:8]: On 2nd thought, see https://github.com/conda/conda-libmamba-solver/pull/187#issuecomment-1509764952

That would imply reverting this change.

**Comment by jaimergp** [comment]
More data:

- `tests/cli/test_main_clean.py::test_clean_all_mock_lstat[False]`. Fixed. We had some strange logic for json/quiet/verbose combos that didn't work for this particular test.
- `tests/core/test_subdir_data.py::test_state_is_not_json` only fails on Windows? 🤔 
- `tests/test_activate.py::ShellWrapperIntegrationTests::test_dash_basic_integration` is working now.

**Comment by jaimergp** [comment]
Cannot reproduce `tests/core/test_subdir_data.py::test_state_is_not_json` on the Windows VM either :/

File: dev/requirements.txt (hunk 2 above)
Comment (costrouc) [reply dev/requirements.txt:8]: Should we apply this change? I'm worried this will immediately start failing when 1.4.3 is released.

Comment (jaimergp) [reply dev/requirements.txt:8]: I worry that if we lock the version like that, other problems arise in future releases and we don't see them.

File: news/ci-failure-187 [hunk 4]
```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* CI failures
```
Comment (jaimergp) [inline news/ci-failure-187:7]:
```suggestion
* Do not set `quiet` manually when `context.json` is true. (#187)
```

File: news/ci-failure-187 [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+* <news item>
+
+### Docs
+
+* <news item>
+
+### Other
+
+* <news item>
```
Comment (jaimergp) [inline news/ci-failure-187:19]:
```suggestion
* Skip tests on libmamba 1.4.2 temporarily to workaround some test failures. Tracked by #186. (#187)
```

**Review by jaimergp — APPROVED** [review]
